### PR TITLE
feat: added timer, search, share, knowledge base and create quiz

### PR DESCRIPTION
**Pull request by Chakravarthi0**
Implemented the following features:
- timer
- search
- share 
- knowledge base
- create quiz

Timer: user has to complete the quiz within 1 minute and 30 seconds.

Search: user can search from quizzes page

Share: user can share the URL of a particular quiz.

Knowledge base: user can access their stats from the profile page. User's game stats includes, number of games attended, total points gained so far, knowledge level.

Create quiz: user can create quiz under a particular category. Each quiz can contain 5-10 questions.

### CODE REVIEW

File: src/components/knowledge-board/knowledge-board.css [hunk 1]
```diff
@@ -0,0 +1,36 @@
+.vl {
```
Comment (msdeshmukh009) [inline src/components/knowledge-board/knowledge-board.css]: Naming can be better here.

File: src/components/question/Question.tsx [hunk 2]
```diff
@@ -8,75 +9,114 @@ const Question = () => {
   const [selectedOptionIndex, setSelectedOptionIndex] = useState(-1);
   const { getNextQuestion, getPreviousQuestion, gameState, setOption } =
     useGame();
-  const { questions, currentQuestionIndex, selectedOptions } = gameState;
-  const { question, options } = questions[currentQuestionIndex];
+  console.log(gameState);
```
Comment (msdeshmukh009) [inline src/components/question/Question.tsx]: You can remove this console

File: src/components/quiz-card/QuizCard.tsx [hunk 3]
```diff
@@ -10,33 +10,33 @@ const QuizCard = ({
   imgUrl,
   id,
 }: quizCardType) => {
-  const { getQuestions } = useGame();
   const navigate = useNavigate();
+
+  const copyUrlTOClipboard = async () => {
+    await navigator.clipboard.writeText(`http://localhost:3000/rules/${id}`);
+    toast.success("Link copied to clipboard");
```
Comment (msdeshmukh009) [inline src/components/quiz-card/QuizCard.tsx]: You need to put the actual URL here.

**Review by msdeshmukh009 — APPROVED** [review]
The code looks fine.
The create quiz feature is working as expected.
Nice work!!

**Comment by Chakravarthi0** [comment]
> The code looks fine. The create quiz feature is working as expected. Nice work!!

Thanks for the review, I have made the suggested changes

File: src/hooks/useQuiz.ts [hunk 4]
```diff
@@ -1,10 +1,22 @@
...
 
-  return { quizState };
+  const refreshQuiz = async () => {
+    quizDispatch({ type: "INITIALIZE" });
+    try {
+      const categories = await getCategories();
+      const quizzes = await getQuizzes();
+      quizDispatch({ type: "SET_QUIZ", payload: { categories, quizzes } });
+    } catch (err) {
+      console.log(err);
```
Comment (manojasarsa) [inline src/hooks/useQuiz.ts:15]: you can use `console.error` here

**Review by manojasarsa — APPROVED** [review]
Code looks clean and self-explanatory.
Nice use of component names.
Types are been handled efficiently.

**Pull request merged**